### ISSUE TITLE
Please reconsider any CI check that requires a maintainer to act on the PR.

### ISSUE THREAD

**Issue by Carreau** [napari/napari#6042]
I find it extremely frustrating to have 2 CI items that makes almost every PRs have red CI status. 
Especially when looking at the PRs list, many PRs are "failing" or "pending" either from the "label required" or "milestone required".

I spend quite some time going through the PR list and try to restart failing CI or fix trivial errors, this make the CI status mark way less useful. 

Moreover I labels and milestone can be added after the PR is merged, so I don't think it is critical to have it as a CI status.

Another possibility is to find a way for the labels and milestones to be automatically added.

[Edit]

It is also impossible for a contributor to act these failing statuses, which I think is a clear indicator that there is a problem with the approach. I think a first time contributor should be able to submit a PR that passes all the checks without any human involvement.

**Comment by Carreau** [comment]
May I suggest then that 1) Tagging should not be left to the release manager, that 2) Instead of CI we have a self-updating pinned issue that list all the PRs that don't have a milestone /label ?

**Comment by Czaki** [comment]
@Carreau I agree that It is not an optimal solution, but I need more time to investigate it as I need to fix missing labels and milestones. 

I totally agree that we need something better. Could you point good examples of projects using merge bots?

**Comment by Carreau** [comment]
The main project I know that uses bots to merge/manage PRs is Rust with bors https://github.com/rust-lang/bors.

One thing that come to mind if we wish to have a merge/squash bot, would be to have a COMMIT_MSG special file that the bot use as the commit msg. That would let us review the commit message themself instead of having it be edited by the maintainer at merge time.

**Comment by Czaki** [comment]
> One thing that come to mind if we wish to have a merge/squash bot, would be to have a COMMIT_MSG special file that the bot use as the commit msg. That would let us review the commit message themself instead of having it be edited by the maintainer at merge time.

I meet this approach during contributing to another OSS, but as it simplifies maintainability, It make a little more maintenance job for contributions from newcomers.

**Comment by Carreau** [comment]
> I meet this approach during contributing to another OSS, but as it simplifies maintainability, It make a little more maintenance job for contributions from newcomers.

That is understandable. Do you remember what the main issue was ? Having users create the commit message file I guess ?

I don't advocate for it to be the _only_ way, and I'm wondering if we could manage that ? Like if the branch does not have a COMMIT_MSG the merge bot could push a commit with this file.

**Comment by Czaki** [comment]
> That is understandable. Do you remember what the main issue was ? Having users create the commit message file I guess ?

Maintainer needs to ask the contributor to add it or do it from hi side. If the user add it then I observe some problems with the file format (but the selection of a good merge bot/release notes tool may solve this). 

Everything to solve, but should not be introduced without preparation of good documentation that lead us to https://github.com/napari/docs/issues/86
